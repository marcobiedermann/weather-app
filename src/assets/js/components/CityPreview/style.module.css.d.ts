declare const styles: {
  readonly city: string;
  readonly 'city--cold': string;
  readonly 'city--warm': string;
};
export = styles;
