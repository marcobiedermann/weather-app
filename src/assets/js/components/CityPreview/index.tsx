import classNames from 'classnames';
import React, { FC } from 'react';
import { Link } from 'react-router-dom';
import styles from './style.module.css';

export interface CityPreviewProps {
  id: number;
  main: {
    temp: number;
  };
  name: string;
  sys: {
    country: string;
  };
}

const CityPreview: FC<CityPreviewProps> = (props) => {
  const { id, main, name, sys, ...otherProps } = props;

  return (
    <div
      className={classNames(styles.city, {
        [styles['city--cold']]: main.temp <= 0,
        [styles['city--warm']]: main.temp >= 15,
      })}
      {...otherProps}
    >
      <Link to={`/cities/${id}`} className={styles.city__link}>
        <div className={styles.city__temp}>{main.temp} Â°</div>
        <h2 className={styles.city__name}>{name}</h2>
        <h3 className={styles.city__country}>{sys.country}</h3>
      </Link>
    </div>
  );
};

export default CityPreview;
