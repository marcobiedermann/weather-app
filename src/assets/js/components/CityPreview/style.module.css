/* stylelint-disable block-no-empty */

.city {
  background-color: #929fb3;
  border-radius: 1.5em;
  padding: 1em;
}

.city > :first-child {
  margin-top: 0;
}

.city > :last-child {
  margin-bottom: 0;
}

.city--cold {
  background-color: #3f73be;
}

.city--warm {
  background-color: #ec9414;
}

.city__country {
  color: #b6bbc7;
  margin: 0;
}

.city__link {
  display: block;
}

.city__name {
  margin: 0;
}

.city__temp {
}
