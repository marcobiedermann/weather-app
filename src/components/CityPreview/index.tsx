import clsx from 'clsx';
import React from 'react';
import { Link } from 'react-router-dom';
import styles from './style.module.css';

interface CityPreviewProps {
  id: number;
  main: {
    temp: number;
  };
  name: string;
  sys: {
    country: string;
  };
}

function CityPreview(props: CityPreviewProps): JSX.Element {
  const { id, main, name, sys, ...otherProps } = props;

  return (
    <div
      className={clsx(styles.city, {
        [styles['city--cold']]: main.temp <= 0,
        [styles['city--warm']]: main.temp >= 15,
      })}
      {...otherProps}
    >
      <Link to={`/cities/${id}`} className={styles.city__link}>
        <div className={styles.city__temp}>{main.temp} Â°</div>
        <h2 className={styles.city__name}>{name}</h2>
        <h3 className={styles.city__country}>{sys.country}</h3>
      </Link>
    </div>
  );
}

export type { CityPreviewProps };
export default CityPreview;
