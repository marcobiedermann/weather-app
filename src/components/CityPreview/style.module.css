.city {
  align-items: center;
  background-color: #fcfcfc;
  border-radius: 0.25em;
  color: #8f9298;
  display: flex;
  justify-content: space-between;
  padding: 1em 1.5em;
  text-decoration: none;
}

.city__country {
  color: inherit;
  font-size: 1.25rem;
  font-weight: normal;
  line-height: 1.25;
  margin-bottom: 0.25em;
  margin-top: 0;
}

.city__name {
  margin-bottom: 0;
  margin-top: 0.25em;
}
