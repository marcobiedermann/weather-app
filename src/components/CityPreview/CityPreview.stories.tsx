import type { Meta, StoryObj } from '@storybook/react';
import CityPreview from '.';

const meta: Meta<typeof CityPreview> = {
  component: CityPreview,
  args: {
    sys: {
      country: 'DE',
    },
    main: {
      temp: 18.54,
    },
    id: 2950159,
    name: 'Berlin',
  },
};

type Story = StoryObj<typeof CityPreview>;

const Default: Story = {};

export { Default };
export default meta;
