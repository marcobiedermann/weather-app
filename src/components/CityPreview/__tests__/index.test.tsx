import { render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { describe, expect, it } from 'vitest';
import CityPreview from '..';
import cityFixtures from '../../City/__fixtures__';

describe('components/CityPreview', () => {
  it('renders correctly', () => {
    expect.assertions(1);

    const { asFragment } = render(
      <MemoryRouter>
        <CityPreview {...cityFixtures} />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
