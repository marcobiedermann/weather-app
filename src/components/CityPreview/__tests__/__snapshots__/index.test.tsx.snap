// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/CityPreview > renders correctly 1`] = `
<DocumentFragment>
  <a
    class="city"
    href="/cities/2950159"
  >
    <div>
      <h3
        class="city__country"
      >
        DE
      </h3>
      <h2
        class="city__name"
      >
        Berlin
      </h2>
    </div>
    <div
      class="city__temp"
    >
      19Â°C
    </div>
  </a>
</DocumentFragment>
`;
